Reduce build times by refactoring WebPage.h

darinadler opened the pull request:
#### 2aea878bc5fd3526dac0cde489c35a5a208b4731
<pre>
Reduce build times by refactoring WebPage.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=256424">https://bugs.webkit.org/show_bug.cgi?id=256424</a>
rdar://108997618

Reviewed by NOBODY (OOPS!).

* Source/JavaScriptCore/inspector/InspectorFrontendChannel.h:
Moved FrontendChannel::ConnectionType out to the namespace level
so it can be forward declared.

* Source/WTF/wtf/RunLoop.cpp:
(WTF::runLoopHolder): Made this a non-member function so we don&apos;t have to include
ThreadSpecific.h in the header.

* Source/WebCore/dom/DOMImplementation.cpp:
(WebCore::DOMImplementation::createDocument): Use AllPlugins and
OnlyApplicationPlugins.

* Source/WebCore/page/MediaControlsContextMenuItem.h: Moved
MediaControlsContextMenuItem::ID out to the namespace level so
it can be fowrad declared.

* Source/WebCore/page/PageOverlay.cpp:
(WebCore::generatePageOverlayID): Use PageOverlayID.

* Source/WebCore/page/PageOverlay.h: Move PageOverlay::PageOverlayID
out to the namespace level so it can be forward declared.

* Source/WebCore/page/WebCoreKeyboardUIMode.h: Add an underlying type
to KeyboardUIMode so it can be forward declared.

* Source/WebCore/plugins/PluginData.cpp:
(WebCore::supportsMimeTypeForPlugins): Use AllowedPluginTypes and AllPlugins.

* Source/WebCore/plugins/PluginData.h: Moved PluginData::AllowedPluginTypes
out to the namespace level so it can be forward declared.

* Source/WebKit/Shared/ContentWorldShared.h: Use a struct for the argument
to ObjectIdentifier so ContentWorldIdentifier can be forward declared.

* Source/WebKit/Shared/WebHitTestResultData.h: Use WebCore::PageOverlayID.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm: Updated includes.
* Source/WebKit/UIProcess/WebPageProxy.cpp: Ditto.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Added WebPageInternals.h
and DrawingAreaInlines.h.

* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIEventCocoa.mm: Updated includes.
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIWebNavigationEventCocoa.mm: Ditto.
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIEvent.h: Ditto. Ditto.
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIWebNavigationEvent.h: Ditto.
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInFrame.mm: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextController.mm: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextControllerInternal.h: Ditto.
* Source/WebKit/WebProcess/Model/ARKitInlinePreviewModelPlayer.h: Ditto.

* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::WebLoaderStrategy::webResourceLoaderForIdentifier const): Moved here from header.
(WebKit::WebLoaderStrategy::ongoingLoads const): Ditto.
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.h: Updated for the above.

* Source/WebKit/WebProcess/WebAuthentication/WebAuthenticatorCoordinator.cpp:
(WebKit::WebAuthenticatorCoordinator::processingUserGesture): Pass nullopt for the
ResourceLoaderIdentifier for addConsoleMessage.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp: Updated includes.
* Source/WebKit/WebProcess/WebCoreSupport/WebDeviceOrientationUpdateProvider.h: Ditto.
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.cpp: Ditto.

* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp: Updated includes.
(WebKit::WebFrameLoaderClient::objectContentType): Use AllPlugins and OnlyApplicationPlugins.

* Source/WebKit/WebProcess/WebCoreSupport/WebProgressTrackerClient.cpp: Updated includes.

* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm: Updated includes.
(WebKit::OverridePasteboardForSelectionReplacement::OverridePasteboardForSelectionReplacement):
Use Span instead of IPC::DataReference.
(WebKit::WebPage::replaceImageForRemoveBackground): Ditto.
(WebKit::WebPage::replaceSelectionWithPasteboardData): Ditto.
(WebKit::WebPage::sanitizeLookalikeCharacters): Moved non-Apple-internal version to WebPage.cpp.
(WebKit::WebPage::allowedLookalikeCharacters): Ditto.

* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
Updated includes.

* Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
(WebKit::DrawingArea::displayColorSpace const): Moved here from header.
(WebKit::DrawingArea::activityStateDidChange): Ditto.
(WebKit::DrawingArea::setColorSpace): Ditto.
(WebKit::DrawingArea::acceleratedAnimationDidStart): Ditto.
(WebKit::DrawingArea::acceleratedAnimationDidEnd): Ditto.
(WebKit::DrawingArea::adjustTransientZoom): Ditto.
(WebKit::DrawingArea::commitTransientZoom): Ditto.
(WebKit::DrawingArea::addRootFrame): Ditto.
(WebKit::DrawingArea::mainFrameContentSizeChanged): Ditto.
(WebKit::DrawingArea::attachViewOverlayGraphicsLayer): Ditto.

* Source/WebKit/WebProcess/WebPage/DrawingArea.h: Reduced includes,
added forward declarations, moved functions out of the header that
can&apos;t compile without the includes. These were virtual functions and
not inlined in practice anyway except for send, which is moved into
DrawingAreaInlines.h.

* Source/WebKit/WebProcess/WebPage/DrawingAreaInlines.h: Added.

* Source/WebKit/WebProcess/WebPage/EventDispatcher.h: Moved
EventDispatcher::WheelEventOrigin out to the namespace level so it can
be forward declared.

* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDisplayRefreshMonitor.h:
Updated includes.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h: Ditto.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm: Ditto.
* Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp: Ditto.

* Source/WebKit/WebProcess/WebPage/WebBackForwardListProxy.h: Moved
WebBackForwardListProxy::OverwriteExistingItem out to the namespace
level so it cna be forward declared.

* Source/WebKit/WebProcess/WebPage/WebContextMenu.cpp: Updated includes.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp: Updated includes.
(WebKit::WebPage::Internals::Internals): Added.
(WebKit::overriddenMediaType): Updated to use internals().
(WebKit::WebPage::WebPage): Ditto.
(WebKit::m_appHighlightsVisible): Ditto.
(WebKit::WebPage::reinitializeWebPage): Ditto.
(WebKit::WebPage::updateThrottleState): Ditto.
(WebKit::WebPage::isThrottleable const): Ditto.
(WebKit::WebPage::initializeInjectedBundleFullScreenClient): Ditto.
(WebKit::WebPage::editorState const): Ditto.
(WebKit::WebPage::close): Ditto.
(WebKit::WebPage::loadDataInFrame): Ditto.
(WebKit::WebPage::loadRequest): Ditto.
(WebKit::WebPage::loadDataImpl): Ditto.
(WebKit::WebPage::loadAlternateHTML): Ditto.
(WebKit::WebPage::loadSimulatedRequestAndResponse): Ditto.
(WebKit::WebPage::goToBackForwardItem): Ditto.
(WebKit::WebPage::setSize): Ditto.
(WebKit::WebPage::sendViewportAttributesChanged): Ditto.
(WebKit::WebPage::scalePage): Ditto.
(WebKit::WebPage::setFixedLayoutSize): Ditto.
(WebKit::WebPage::setDefaultUnobscuredSize): Ditto.
(WebKit::WebPage::updateSizeForCSSDefaultViewportUnits): Ditto.
(WebKit::WebPage::setMinimumUnobscuredSize): Ditto.
(WebKit::WebPage::updateSizeForCSSSmallViewportUnits): Ditto.
(WebKit::WebPage::setMaximumUnobscuredSize): Ditto.
(WebKit::WebPage::updateSizeForCSSLargeViewportUnits): Ditto.
(WebKit::WebPage::disabledAdaptationsDidChange): Ditto.
(WebKit::WebPage::viewportPropertiesDidChange): Ditto.
(WebKit::WebPage::pageDidScroll): Ditto.
(WebKit::WebPage::freezeLayerTree): Ditto.
(WebKit::WebPage::unfreezeLayerTree): Ditto.
(WebKit::WebPage::isLayerTreeFrozen): Ditto.
(WebKit::WebPage::updateDrawingAreaLayerTreeFreezeState): Ditto.
(WebKit::WebPage::markLayersVolatile): Ditto.
(WebKit::WebPage::tryMarkLayersVolatileCompletionHandler): Ditto.
(WebKit::WebPage::cancelMarkLayersVolatile): Ditto.
(WebKit::WebPage::contextMenuForKeyEvent): Ditto.
(WebKit::WebPage::mouseEvent): Ditto.
(WebKit::WebPage::handleWheelEvent): Ditto.
(WebKit::WebPage::wheelEvent): Ditto.
(WebKit::WebPage::keyEvent): Ditto.
(WebKit::WebPage::restoreSessionInternal): Ditto.
(WebKit::WebPage::restoreSession): Ditto.
(WebKit::WebPage::updateBackForwardListForReattach): Ditto.
(WebKit::WebPage::setCurrentHistoryItemForReattach): Ditto.
(WebKit::WebPage::cancelCurrentInteractionInformationRequest): Ditto.
(WebKit::WebPage::dispatchTouchEvent): Ditto.
(WebKit::WebPage::setBackgroundColor): Ditto.
(WebKit::WebPage::updateIsInWindow): Ditto.
(WebKit::WebPage::visibilityDidChange): Ditto.
(WebKit::WebPage::setActivityState): Ditto.
(WebKit::WebPage::runJavaScript): Ditto.
(WebKit::WebPage::runJavaScriptInFrameInScriptWorld): Ditto.
(WebKit::WebPage::willCommitLayerTree): Ditto.
(WebKit::WebPage::didFlushLayerTreeAtTime): Ditto.
(WebKit::WebPage::performDragControllerAction): Ditto.
(WebKit::WebPage::webUndoStep): Ditto.
(WebKit::WebPage::addWebUndoStep): Ditto.
(WebKit::WebPage::removeWebEditCommand): Ditto.
(WebKit::WebPage::didChooseFilesForOpenPanelWithDisplayStringAndIcon): Ditto.
(WebKit::WebPage::updateMainFrameScrollOffsetPinning): Ditto.
(WebKit::WebPage::mainFrameDidLayout): Ditto.
(WebKit::WebPage::windowAndViewFramesChanged): Ditto.
(WebKit::WebPage::addResourceRequest): Ditto.
(WebKit::WebPage::removeResourceRequest): Ditto.
(WebKit::WebPage::setMayStartMediaWhenInWindow): Ditto.
(WebKit::WebPage::elementDidFocus): Ditto.
(WebKit::WebPage::setMinimumSizeForAutoLayout): Ditto.
(WebKit::WebPage::setSizeToContentAutoSizeMaximumSize): Ditto.
(WebKit::WebPage::setAutoSizingShouldExpandToViewHeight): Ditto.
(WebKit::WebPage::setViewportSizeForCSSViewportUnits): Ditto.
(WebKit::WebPage::canShowMIMEType const): Ditto.
(WebKit::WebPage::addTextCheckingRequest): Ditto.
(WebKit::WebPage::didFinishCheckingText): Ditto.
(WebKit::WebPage::didCancelCheckingText): Ditto.
(WebKit::WebPage::willReplaceMultipartContent): Ditto.
(WebKit::WebPage::didReplaceMultipartContent): Ditto.
(WebKit::WebPage::didCommitLoad): Ditto.
(WebKit::WebPage::updateWebsitePolicies): Ditto.
(WebKit::WebPage::createDocumentLoader): Ditto.
(WebKit::WebPage::updateIntrinsicContentSizeIfNeeded): Ditto.
(WebKit::WebPage::flushPendingIntrinsicContentSizeUpdate): Ditto.
(WebKit::WebPage::scheduleIntrinsicContentSizeUpdate): Ditto.
(WebKit::WebPage::registerURLSchemeHandler): Ditto.
(WebKit::WebPage::urlSchemeTaskWillPerformRedirection): Ditto.
(WebKit::WebPage::urlSchemeTaskDidPerformRedirection): Ditto.
(WebKit::WebPage::urlSchemeTaskDidReceiveResponse): Ditto.
(WebKit::WebPage::urlSchemeTaskDidReceiveData): Ditto.
(WebKit::WebPage::urlSchemeTaskDidComplete): Ditto.
(WebKit::WebPage::hasStorageAccess): Ditto.
(WebKit::WebPage::requestStorageAccess): Ditto.
(WebKit::WebPage::addDomainWithPageLevelStorageAccess): Ditto.
(WebKit::WebPage::hasPageLevelStorageAccess const): Ditto.
(WebKit::WebPage::clearPageLevelStorageAccess): Ditto.
(WebKit::WebPage::didLoadFromRegistrableDomain): Ditto.
(WebKit::WebPage::getLoadedSubresourceDomains): Ditto.
(WebKit::WebPage::clearLoadedSubresourceDomains): Ditto.
(WebKit::WebPage::elementForContext const): Ditto.
(WebKit::WebPage::contextForElement const): Ditto.
(WebKit::WebPage::startTextManipulations): Ditto.
(WebKit::WebPage::startTextManipulationForFrame): Ditto.
(WebKit::WebPage::textAutosizingUsesIdempotentModeChanged): Ditto.
(WebKit::WebPage::synchronizeCORSDisablingPatternsWithNetworkProcess): Ditto.
(WebKit::WebPage::createAppHighlightInSelectedRange): Ditto.
(WebKit::WebPage::shouldAllowRemoveBackground const): Ditto.
(WebKit::WebPage::setIsWindowResizingEnabled): Ditto.
(WebKit::WebPage::setLookalikeCharacterStrings): Ditto.
(WebKit::WebPage::setAllowedLookalikeCharacterStrings): Ditto.
(WebKit::WebPage::injectedBundleFullScreenClient): Added.
(WebKit::WebPage::beginPrintingDuringDOMPrintOperation): Added.
(WebKit::WebPage::drawPagesToPDFDuringDOMPrintOperation): Added.
(WebKit::WebPage::drawRectToImageDuringDOMPrintOperation): Added.
(WebKit::WebPage::computePagesForPrintingDuringDOMPrintOperation): Added.
(WebKit::WebPage::drawPagesForPrintingDuringDOMPrintOperation): Added.
(WebKit::WebPage::identifier const): Added.
(WebKit::WebPage::activityState const): Added.
(WebKit::WebPage::underlayColor const): Added.
(WebKit::WebPage::setUnderlayColor): Added.
(WebKit::WebPage::backgroundColor const): Added.
(WebKit::WebPage::highlightIsNewGroup const): Added.
(WebKit::WebPage::accessibilityPosition const): Added.
(WebKit::WebPage::webPageProxyIdentifier const): Added.
(WebKit::WebPage::allowedDragSourceActions const): Added.
(WebKit::WebPage::loadedSubresourceDomains const): Added.
(WebKit::WebPage::minimumSizeForAutoLayout const): Added.
(WebKit::WebPage::highlightRequestOriginatedInApp const): Added.
(WebKit::WebPage::viewportSizeForCSSViewportUnits const): Added.
(WebKit::WebPage::sizeToContentAutoSizeMaximumSize const): Added.
(WebKit::WebPage::windowFrameInUnflippedScreenCoordinates const): Added.
(WebKit::WebPage::size const): Added.
(WebKit::WebPage::sessionStorageNamespaceIdentifier const): Added.
(WebKit::WebPage::isVisible const): Added.
(WebKit::WebPage::isVisibleOrOccluded const): Added.
(WebKit::WebPage::viewportConfiguration const): Added.
(WebKit::WebPage::hostFileDescriptor const): Added.
(WebKit::WebPage::sanitizeLookalikeCharacters): Added.
(WebKit::WebPage::allowedLookalikeCharacters): Added.

* Source/WebKit/WebProcess/WebPage/WebPage.h: Removed most includes.
Added forward declarations. Moved things that required the includes
into WebPageInternals.h, WebPageInlines.h, and WebPage.cpp.

* Source/WebKit/WebProcess/WebPage/WebPageInlines.h: Moved
userContentControllerIdentifier and sendSyncWithDelayedReply here.
Moved sessionStorageNamespaceIdentifier, isVisible, and
isVisibleOrOccluded into WebPage.cpp.

* Source/WebKit/WebProcess/WebPage/WebPageInternals.h: Added.

* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::inputMethodState): Renamed.
(WebKit::WebPage::setInputMethodState): Use internals().
(WebKit::WebPage::sendMessageToWebProcessExtensionWithReply): Ditto.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::screenSize const): Use internals().
(WebKit::WebPage::availableScreenSize const): Ditto.
(WebKit::WebPage::overrideScreenSize const): Ditto.
(WebKit::WebPage::savePageState): Ditto.
(WebKit::WebPage::restorePageState): Ditto.
(WebKit::WebPage::minimumPageScaleFactor const): Ditto.
(WebKit::WebPage::maximumPageScaleFactor const): Ditto.
(WebKit::WebPage::maximumPageScaleFactorIgnoringAlwaysScalable const): Ditto.
(WebKit::WebPage::allowsUserScaling const): Ditto.
(WebKit::WebPage::handleEditingKeyboardEvent): Ditto.
(WebKit::WebPage::registerUIProcessAccessibilityTokens): Ditto.
(WebKit::WebPage::rectForElementAtInteractionLocation const): Ditto.
(WebKit::WebPage::handleSyntheticClick): Ditto.
(WebKit::WebPage::didFinishContentChangeObserving): Ditto.
(WebKit::WebPage::completeSyntheticClick): Ditto.
(WebKit::WebPage::attemptSyntheticClick): Ditto.
(WebKit::WebPage::requestDragStart): Ditto.
(WebKit::WebPage::requestAdditionalItemsForDragSession): Ditto.
(WebKit::WebPage::didConcludeDrop): Ditto.
(WebKit::WebPage::didConcludeEditDrag): Ditto.
(WebKit::WebPage::didFinishLoadingImageForElement): Ditto.
(WebKit::WebPage::computeAndSendEditDragSnapshot): Ditto.
(WebKit::WebPage::handleTwoFingerTapAtPoint): Ditto.
(WebKit::WebPage::potentialTapAtPosition): Ditto.
(WebKit::WebPage::commitPotentialTap): Ditto.
(WebKit::WebPage::commitPotentialTapFailed): Ditto.
(WebKit::WebPage::cancelPotentialTapInFrame): Ditto.
(WebKit::WebPage::setForceAlwaysUserScalable): Ditto.
(WebKit::WebPage::selectWithGesture): Ditto.
(WebKit::WebPage::clearSelection): Ditto.
(WebKit::WebPage::getRectsForGranularityWithSelectionOffset): Ditto.
(WebKit::WebPage::storeSelectionForAccessibility): Ditto.
(WebKit::WebPage::getRectsAtSelectionOffsetWithText): Ditto.
(WebKit::WebPage::setSelectionRange): Ditto.
(WebKit::WebPage::updateSelectionWithExtentPointAndBoundary): Ditto.
(WebKit::WebPage::focusedElementInformation): Ditto.
(WebKit::WebPage::setViewportConfigurationViewLayoutSize): Ditto.
(WebKit::WebPage::dynamicViewportSizeUpdate): Ditto.
(WebKit::WebPage::resetViewportDefaultConfiguration): Ditto.
(WebKit::WebPage::resetIdempotentTextAutosizingIfNeeded): Ditto.
(WebKit::WebPage::shrinkToFitContent): Ditto.
(WebKit::WebPage::viewportConfigurationChanged): Ditto.
(WebKit::WebPage::applicationDidFinishSnapshottingAfterEnteringBackground): Ditto.
(WebKit::WebPage::scaleFromUIProcess const): Ditto.
(WebKit::WebPage::updateVisibleContentRects): Ditto.
(WebKit::WebPage::textInputContextsInRect): Ditto.
(WebKit::WebPage::platformDidScalePage): Ditto.

* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.cpp:
Updated includes.

* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
Updated includes.

* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::executeKeypressCommandsInternal): Use internals().
(WebKit::WebPage::registerUIProcessAccessibilityTokens): Ditto.
(WebKit::WebPage::zoomPDFIn): Ditto.
(WebKit::WebPage::zoomPDFOut): Ditto.
(WebKit::WebPage::savePDF): Ditto.
(WebKit::WebPage::openPDFWithPreview): Ditto.
(WebKit::WebPage::createPDFHUD): Ditto.
(WebKit::WebPage::updatePDFHUDLocation): Ditto.
(WebKit::WebPage::removePDFHUD): Ditto.

* Source/WebKit/WebProcess/WebProcess.cpp: Updated includes.

* Source/WebKit/WebProcess/WebStorage/StorageNamespaceIdentifier.h:
Use struct for StorageNamespaceIdentifierType so it can be
forward declared.

* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.mm:
Updated includes.

* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _canProvideDocumentSource]): Use WebCore::AllPlugins
and WebCore::OnlyApplicationPlugins.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2aea878bc5fd3526dac0cde489c35a5a208b4731

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8112 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8399 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8616 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9770 "Built successfully") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8199 "Failed to compile WebKit") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/8118 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10386 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8310 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11063 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8256 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9334 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7380 "1 api test failed or timed out") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9891 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6635 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7426 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14998 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/6918 "Built successfully and passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7758 "1 api test failed or timed out") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7551 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10925 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/7674 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8019 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6521 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/8282 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7324 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1890 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11532 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/8504 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7762 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2040 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on Source/WebKit/WebProcess/WebPage/WebPage.h:
```diff
@@ -695,20 +709,20 @@ class WebPage : public API::ObjectImpl<API::Object::Type::BundlePage>, public IP
     void setLayerHostingMode(LayerHostingMode);
 
 #if PLATFORM(COCOA)
     void updatePluginsActiveAndFocusedState();
```
@darinadler - I was looking into different stuff but noticed that this is unused and don't have any reference. So I noticed that you are doing this work, so rather than doing another PR, I thought to mention so you can get rid of it as well.

Ahmad-S792 replied:
Sorry - I am not super familiar with GitHub review tool. I am referring to this: `void updatePluginsActiveAndFocusedState();`.

Ahmad-S792 replied:
<img width="685" alt="image" src="https://github.com/WebKit/WebKit/assets/52317531/4906714b-66ca-4f3b-855c-2dce036acc01">
It  has only this reference here and not used anywhere else in WebKit code.

darinadler replied:
Yes, I’ll remove `updatePluginsActiveAndFocusedState`.

darinadler committed:
Reduce build times by refactoring WebPage.h
https://bugs.webkit.org/show_bug.cgi?id=256424
rdar://108997618

Reviewed by NOBODY (OOPS!).

* Source/JavaScriptCore/inspector/InspectorFrontendChannel.h:
Moved FrontendChannel::ConnectionType out to the namespace level
so it can be forward declared.

* Source/WTF/wtf/RunLoop.cpp:
(WTF::runLoopHolder): Made this a non-member function so we don't have to include
ThreadSpecific.h in the header.

* Source/WebCore/dom/DOMImplementation.cpp:
(WebCore::DOMImplementation::createDocument): Use AllPlugins and
OnlyApplicationPlugins.

* Source/WebCore/page/MediaControlsContextMenuItem.h: Moved
MediaControlsContextMenuItem::ID out to the namespace level so
it can be fowrad declared.

* Source/WebCore/page/PageOverlay.cpp:
(WebCore::generatePageOverlayID): Use PageOverlayID.

* Source/WebCore/page/PageOverlay.h: Move PageOverlay::PageOverlayID
out to the namespace level so it can be forward declared.

* Source/WebCore/page/WebCoreKeyboardUIMode.h: Add an underlying type
to KeyboardUIMode so it can be forward declared.

* Source/WebCore/plugins/PluginData.cpp:
(WebCore::supportsMimeTypeForPlugins): Use AllowedPluginTypes and AllPlugins.

* Source/WebCore/plugins/PluginData.h: Moved PluginData::AllowedPluginTypes
out to the namespace level so it can be forward declared.

* Source/WebKit/Shared/ContentWorldShared.h: Use a struct for the argument
to ObjectIdentifier so ContentWorldIdentifier can be forward declared.

* Source/WebKit/Shared/WebHitTestResultData.h: Use WebCore::PageOverlayID.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm: Updated includes.
* Source/WebKit/UIProcess/WebPageProxy.cpp: Ditto.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Added WebPageInternals.h
and DrawingAreaInlines.h.

* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIEventCocoa.mm: Updated includes.
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIWebNavigationEventCocoa.mm: Ditto.
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIEvent.h: Ditto. Ditto.
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIWebNavigationEvent.h: Ditto.
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInFrame.mm: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextController.mm: Ditto.
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKWebProcessPlugInBrowserContextControllerInternal.h: Ditto.
* Source/WebKit/WebProcess/Model/ARKitInlinePreviewModelPlayer.h: Ditto.

* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::WebLoaderStrategy::webResourceLoaderForIdentifier const): Moved here from header.
(WebKit::WebLoaderStrategy::ongoingLoads const): Ditto.
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.h: Updated for the above.

* Source/WebKit/WebProcess/WebAuthentication/WebAuthenticatorCoordinator.cpp:
(WebKit::WebAuthenticatorCoordinator::processingUserGesture): Pass nullopt for the
ResourceLoaderIdentifier for addConsoleMessage.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp: Updated includes.
* Source/WebKit/WebProcess/WebCoreSupport/WebDeviceOrientationUpdateProvider.h: Ditto.
* Source/WebKit/WebProcess/WebCoreSupport/WebEditorClient.cpp: Ditto.

* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp: Updated includes.
(WebKit::WebFrameLoaderClient::objectContentType): Use AllPlugins and OnlyApplicationPlugins.

* Source/WebKit/WebProcess/WebCoreSupport/WebProgressTrackerClient.cpp: Updated includes.

* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm: Updated includes.
(WebKit::OverridePasteboardForSelectionReplacement::OverridePasteboardForSelectionReplacement):
Use Span instead of IPC::DataReference.
(WebKit::WebPage::replaceImageForRemoveBackground): Ditto.
(WebKit::WebPage::replaceSelectionWithPasteboardData): Ditto.
(WebKit::WebPage::sanitizeLookalikeCharacters): Moved non-Apple-internal version to WebPage.cpp.
(WebKit::WebPage::allowedLookalikeCharacters): Ditto.

* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
Updated includes.

* Source/WebKit/WebProcess/WebPage/DrawingArea.cpp:
(WebKit::DrawingArea::displayColorSpace const): Moved here from header.
(WebKit::DrawingArea::activityStateDidChange): Ditto.
(WebKit::DrawingArea::setColorSpace): Ditto.
(WebKit::DrawingArea::acceleratedAnimationDidStart): Ditto.
(WebKit::DrawingArea::acceleratedAnimationDidEnd): Ditto.
(WebKit::DrawingArea::adjustTransientZoom): Ditto.
(WebKit::DrawingArea::commitTransientZoom): Ditto.
(WebKit::DrawingArea::addRootFrame): Ditto.
(WebKit::DrawingArea::mainFrameContentSizeChanged): Ditto.
(WebKit::DrawingArea::attachViewOverlayGraphicsLayer): Ditto.

* Source/WebKit/WebProcess/WebPage/DrawingArea.h: Reduced includes,
added forward declarations, moved functions out of the header that
can't compile without the includes. These were virtual functions and
not inlined in practice anyway except for send, which is moved into
DrawingAreaInlines.h.

* Source/WebKit/WebProcess/WebPage/DrawingAreaInlines.h: Added.

* Source/WebKit/WebProcess/WebPage/EventDispatcher.h: Moved
EventDispatcher::WheelEventOrigin out to the namespace level so it can
be forward declared.

* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDisplayRefreshMonitor.h:
Updated includes.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h: Ditto.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm: Ditto.
* Source/WebKit/WebProcess/WebPage/ViewGestureGeometryCollector.cpp: Ditto.

* Source/WebKit/WebProcess/WebPage/WebBackForwardListProxy.h: Moved
WebBackForwardListProxy::OverwriteExistingItem out to the namespace
level so it cna be forward declared.

* Source/WebKit/WebProcess/WebPage/WebContextMenu.cpp: Updated includes.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp: Updated includes.
(WebKit::WebPage::Internals::Internals): Added.
(WebKit::overriddenMediaType): Updated to use internals().
(WebKit::WebPage::WebPage): Ditto.
(WebKit::m_appHighlightsVisible): Ditto.
(WebKit::WebPage::reinitializeWebPage): Ditto.
(WebKit::WebPage::updateThrottleState): Ditto.
(WebKit::WebPage::isThrottleable const): Ditto.
(WebKit::WebPage::initializeInjectedBundleFullScreenClient): Ditto.
(WebKit::WebPage::editorState const): Ditto.
(WebKit::WebPage::close): Ditto.
(WebKit::WebPage::loadDataInFrame): Ditto.
(WebKit::WebPage::loadRequest): Ditto.
(WebKit::WebPage::loadDataImpl): Ditto.
(WebKit::WebPage::loadAlternateHTML): Ditto.
(WebKit::WebPage::loadSimulatedRequestAndResponse): Ditto.
(WebKit::WebPage::goToBackForwardItem): Ditto.
(WebKit::WebPage::setSize): Ditto.
(WebKit::WebPage::sendViewportAttributesChanged): Ditto.
(WebKit::WebPage::scalePage): Ditto.
(WebKit::WebPage::setFixedLayoutSize): Ditto.
(WebKit::WebPage::setDefaultUnobscuredSize): Ditto.
(WebKit::WebPage::updateSizeForCSSDefaultViewportUnits): Ditto.
(WebKit::WebPage::setMinimumUnobscuredSize): Ditto.
(WebKit::WebPage::updateSizeForCSSSmallViewportUnits): Ditto.
(WebKit::WebPage::setMaximumUnobscuredSize): Ditto.
(WebKit::WebPage::updateSizeForCSSLargeViewportUnits): Ditto.
(WebKit::WebPage::disabledAdaptationsDidChange): Ditto.
(WebKit::WebPage::viewportPropertiesDidChange): Ditto.
(WebKit::WebPage::pageDidScroll): Ditto.
(WebKit::WebPage::freezeLayerTree): Ditto.
(WebKit::WebPage::unfreezeLayerTree): Ditto.
(WebKit::WebPage::isLayerTreeFrozen): Ditto.
(WebKit::WebPage::updateDrawingAreaLayerTreeFreezeState): Ditto.
(WebKit::WebPage::markLayersVolatile): Ditto.
(WebKit::WebPage::tryMarkLayersVolatileCompletionHandler): Ditto.
(WebKit::WebPage::cancelMarkLayersVolatile): Ditto.
(WebKit::WebPage::contextMenuForKeyEvent): Ditto.
(WebKit::WebPage::mouseEvent): Ditto.
(WebKit::WebPage::handleWheelEvent): Ditto.
(WebKit::WebPage::wheelEvent): Ditto.
(WebKit::WebPage::keyEvent): Ditto.
(WebKit::WebPage::restoreSessionInternal): Ditto.
(WebKit::WebPage::restoreSession): Ditto.
(WebKit::WebPage::updateBackForwardListForReattach): Ditto.
(WebKit::WebPage::setCurrentHistoryItemForReattach): Ditto.
(WebKit::WebPage::cancelCurrentInteractionInformationRequest): Ditto.
(WebKit::WebPage::dispatchTouchEvent): Ditto.
(WebKit::WebPage::setBackgroundColor): Ditto.
(WebKit::WebPage::updateIsInWindow): Ditto.
(WebKit::WebPage::visibilityDidChange): Ditto.
(WebKit::WebPage::setActivityState): Ditto.
(WebKit::WebPage::runJavaScript): Ditto.
(WebKit::WebPage::runJavaScriptInFrameInScriptWorld): Ditto.
(WebKit::WebPage::willCommitLayerTree): Ditto.
(WebKit::WebPage::didFlushLayerTreeAtTime): Ditto.
(WebKit::WebPage::performDragControllerAction): Ditto.
(WebKit::WebPage::webUndoStep): Ditto.
(WebKit::WebPage::addWebUndoStep): Ditto.
(WebKit::WebPage::removeWebEditCommand): Ditto.
(WebKit::WebPage::didChooseFilesForOpenPanelWithDisplayStringAndIcon): Ditto.
(WebKit::WebPage::updateMainFrameScrollOffsetPinning): Ditto.
(WebKit::WebPage::mainFrameDidLayout): Ditto.
(WebKit::WebPage::windowAndViewFramesChanged): Ditto.
(WebKit::WebPage::addResourceRequest): Ditto.
(WebKit::WebPage::removeResourceRequest): Ditto.
(WebKit::WebPage::setMayStartMediaWhenInWindow): Ditto.
(WebKit::WebPage::elementDidFocus): Ditto.
(WebKit::WebPage::setMinimumSizeForAutoLayout): Ditto.
(WebKit::WebPage::setSizeToContentAutoSizeMaximumSize): Ditto.
(WebKit::WebPage::setAutoSizingShouldExpandToViewHeight): Ditto.
(WebKit::WebPage::setViewportSizeForCSSViewportUnits): Ditto.
(WebKit::WebPage::canShowMIMEType const): Ditto.
(WebKit::WebPage::addTextCheckingRequest): Ditto.
(WebKit::WebPage::didFinishCheckingText): Ditto.
(WebKit::WebPage::didCancelCheckingText): Ditto.
(WebKit::WebPage::willReplaceMultipartContent): Ditto.
(WebKit::WebPage::didReplaceMultipartContent): Ditto.
(WebKit::WebPage::didCommitLoad): Ditto.
(WebKit::WebPage::updateWebsitePolicies): Ditto.
(WebKit::WebPage::createDocumentLoader): Ditto.
(WebKit::WebPage::updateIntrinsicContentSizeIfNeeded): Ditto.
(WebKit::WebPage::flushPendingIntrinsicContentSizeUpdate): Ditto.
(WebKit::WebPage::scheduleIntrinsicContentSizeUpdate): Ditto.
(WebKit::WebPage::registerURLSchemeHandler): Ditto.
(WebKit::WebPage::urlSchemeTaskWillPerformRedirection): Ditto.
(WebKit::WebPage::urlSchemeTaskDidPerformRedirection): Ditto.
(WebKit::WebPage::urlSchemeTaskDidReceiveResponse): Ditto.
(WebKit::WebPage::urlSchemeTaskDidReceiveData): Ditto.
(WebKit::WebPage::urlSchemeTaskDidComplete): Ditto.
(WebKit::WebPage::hasStorageAccess): Ditto.
(WebKit::WebPage::requestStorageAccess): Ditto.
(WebKit::WebPage::addDomainWithPageLevelStorageAccess): Ditto.
(WebKit::WebPage::hasPageLevelStorageAccess const): Ditto.
(WebKit::WebPage::clearPageLevelStorageAccess): Ditto.
(WebKit::WebPage::didLoadFromRegistrableDomain): Ditto.
(WebKit::WebPage::getLoadedSubresourceDomains): Ditto.
(WebKit::WebPage::clearLoadedSubresourceDomains): Ditto.
(WebKit::WebPage::elementForContext const): Ditto.
(WebKit::WebPage::contextForElement const): Ditto.
(WebKit::WebPage::startTextManipulations): Ditto.
(WebKit::WebPage::startTextManipulationForFrame): Ditto.
(WebKit::WebPage::textAutosizingUsesIdempotentModeChanged): Ditto.
(WebKit::WebPage::synchronizeCORSDisablingPatternsWithNetworkProcess): Ditto.
(WebKit::WebPage::createAppHighlightInSelectedRange): Ditto.
(WebKit::WebPage::shouldAllowRemoveBackground const): Ditto.
(WebKit::WebPage::setIsWindowResizingEnabled): Ditto.
(WebKit::WebPage::setLookalikeCharacterStrings): Ditto.
(WebKit::WebPage::setAllowedLookalikeCharacterStrings): Ditto.
(WebKit::WebPage::injectedBundleFullScreenClient): Added.
(WebKit::WebPage::beginPrintingDuringDOMPrintOperation): Added.
(WebKit::WebPage::drawPagesToPDFDuringDOMPrintOperation): Added.
(WebKit::WebPage::drawRectToImageDuringDOMPrintOperation): Added.
(WebKit::WebPage::computePagesForPrintingDuringDOMPrintOperation): Added.
(WebKit::WebPage::drawPagesForPrintingDuringDOMPrintOperation): Added.
(WebKit::WebPage::identifier const): Added.
(WebKit::WebPage::activityState const): Added.
(WebKit::WebPage::underlayColor const): Added.
(WebKit::WebPage::setUnderlayColor): Added.
(WebKit::WebPage::backgroundColor const): Added.
(WebKit::WebPage::highlightIsNewGroup const): Added.
(WebKit::WebPage::accessibilityPosition const): Added.
(WebKit::WebPage::webPageProxyIdentifier const): Added.
(WebKit::WebPage::allowedDragSourceActions const): Added.
(WebKit::WebPage::loadedSubresourceDomains const): Added.
(WebKit::WebPage::minimumSizeForAutoLayout const): Added.
(WebKit::WebPage::highlightRequestOriginatedInApp const): Added.
(WebKit::WebPage::viewportSizeForCSSViewportUnits const): Added.
(WebKit::WebPage::sizeToContentAutoSizeMaximumSize const): Added.
(WebKit::WebPage::windowFrameInUnflippedScreenCoordinates const): Added.
(WebKit::WebPage::size const): Added.
(WebKit::WebPage::sessionStorageNamespaceIdentifier const): Added.
(WebKit::WebPage::isVisible const): Added.
(WebKit::WebPage::isVisibleOrOccluded const): Added.
(WebKit::WebPage::viewportConfiguration const): Added.
(WebKit::WebPage::hostFileDescriptor const): Added.
(WebKit::WebPage::sanitizeLookalikeCharacters): Added.
(WebKit::WebPage::allowedLookalikeCharacters): Added.

* Source/WebKit/WebProcess/WebPage/WebPage.h: Removed most includes.
Added forward declarations. Moved things that required the includes
into WebPageInternals.h, WebPageInlines.h, and WebPage.cpp.

* Source/WebKit/WebProcess/WebPage/WebPageInlines.h: Moved
userContentControllerIdentifier and sendSyncWithDelayedReply here.
Moved sessionStorageNamespaceIdentifier, isVisible, and
isVisibleOrOccluded into WebPage.cpp.

* Source/WebKit/WebProcess/WebPage/WebPageInternals.h: Added.

* Source/WebKit/WebProcess/WebPage/glib/WebPageGLib.cpp:
(WebKit::inputMethodState): Renamed.
(WebKit::WebPage::setInputMethodState): Use internals().
(WebKit::WebPage::sendMessageToWebProcessExtensionWithReply): Ditto.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::screenSize const): Use internals().
(WebKit::WebPage::availableScreenSize const): Ditto.
(WebKit::WebPage::overrideScreenSize const): Ditto.
(WebKit::WebPage::savePageState): Ditto.
(WebKit::WebPage::restorePageState): Ditto.
(WebKit::WebPage::minimumPageScaleFactor const): Ditto.
(WebKit::WebPage::maximumPageScaleFactor const): Ditto.
(WebKit::WebPage::maximumPageScaleFactorIgnoringAlwaysScalable const): Ditto.
(WebKit::WebPage::allowsUserScaling const): Ditto.
(WebKit::WebPage::handleEditingKeyboardEvent): Ditto.
(WebKit::WebPage::registerUIProcessAccessibilityTokens): Ditto.
(WebKit::WebPage::rectForElementAtInteractionLocation const): Ditto.
(WebKit::WebPage::handleSyntheticClick): Ditto.
(WebKit::WebPage::didFinishContentChangeObserving): Ditto.
(WebKit::WebPage::completeSyntheticClick): Ditto.
(WebKit::WebPage::attemptSyntheticClick): Ditto.
(WebKit::WebPage::requestDragStart): Ditto.
(WebKit::WebPage::requestAdditionalItemsForDragSession): Ditto.
(WebKit::WebPage::didConcludeDrop): Ditto.
(WebKit::WebPage::didConcludeEditDrag): Ditto.
(WebKit::WebPage::didFinishLoadingImageForElement): Ditto.
(WebKit::WebPage::computeAndSendEditDragSnapshot): Ditto.
(WebKit::WebPage::handleTwoFingerTapAtPoint): Ditto.
(WebKit::WebPage::potentialTapAtPosition): Ditto.
(WebKit::WebPage::commitPotentialTap): Ditto.
(WebKit::WebPage::commitPotentialTapFailed): Ditto.
(WebKit::WebPage::cancelPotentialTapInFrame): Ditto.
(WebKit::WebPage::setForceAlwaysUserScalable): Ditto.
(WebKit::WebPage::selectWithGesture): Ditto.
(WebKit::WebPage::clearSelection): Ditto.
(WebKit::WebPage::getRectsForGranularityWithSelectionOffset): Ditto.
(WebKit::WebPage::storeSelectionForAccessibility): Ditto.
(WebKit::WebPage::getRectsAtSelectionOffsetWithText): Ditto.
(WebKit::WebPage::setSelectionRange): Ditto.
(WebKit::WebPage::updateSelectionWithExtentPointAndBoundary): Ditto.
(WebKit::WebPage::focusedElementInformation): Ditto.
(WebKit::WebPage::setViewportConfigurationViewLayoutSize): Ditto.
(WebKit::WebPage::dynamicViewportSizeUpdate): Ditto.
(WebKit::WebPage::resetViewportDefaultConfiguration): Ditto.
(WebKit::WebPage::resetIdempotentTextAutosizingIfNeeded): Ditto.
(WebKit::WebPage::shrinkToFitContent): Ditto.
(WebKit::WebPage::viewportConfigurationChanged): Ditto.
(WebKit::WebPage::applicationDidFinishSnapshottingAfterEnteringBackground): Ditto.
(WebKit::WebPage::scaleFromUIProcess const): Ditto.
(WebKit::WebPage::updateVisibleContentRects): Ditto.
(WebKit::WebPage::textInputContextsInRect): Ditto.
(WebKit::WebPage::platformDidScalePage): Ditto.

* Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.cpp:
Updated includes.

* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
Updated includes.

* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::executeKeypressCommandsInternal): Use internals().
(WebKit::WebPage::registerUIProcessAccessibilityTokens): Ditto.
(WebKit::WebPage::zoomPDFIn): Ditto.
(WebKit::WebPage::zoomPDFOut): Ditto.
(WebKit::WebPage::savePDF): Ditto.
(WebKit::WebPage::openPDFWithPreview): Ditto.
(WebKit::WebPage::createPDFHUD): Ditto.
(WebKit::WebPage::updatePDFHUDLocation): Ditto.
(WebKit::WebPage::removePDFHUD): Ditto.

* Source/WebKit/WebProcess/WebProcess.cpp: Updated includes.

* Source/WebKit/WebProcess/WebStorage/StorageNamespaceIdentifier.h:
Use struct for StorageNamespaceIdentifierType so it can be
forward declared.

* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.mm:
Updated includes.

* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _canProvideDocumentSource]): Use WebCore::AllPlugins
and WebCore::OnlyApplicationPlugins.

brentfulgham commented on Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
```diff
@@ -471,7 +473,7 @@ static String commandNameForSelectorName(const String& selectorName)
 }
 
-void WebPage::registerUIProcessAccessibilityTokens(const IPC::DataReference& elementToken, const IPC::DataReference& windowToken)
+void WebPage::registerUIProcessAccessibilityTokens(std::span<const uint8_t> elementToken, std::span<const uint8_t> windowToken)
```
Are we retaining the semantics of a `const &` by using a `span`? Should we be using a `const span<>` instead?

darinadler replied:
We are. The `const` in `const uint8_t` is what tells us that we can’t change the pointed-to characters.

brentfulgham replied:
Ah, got it. Thank you for clarifying.

brentfulgham commented on Source/WebCore/page/PageOverlay.h:
```diff
@@ -77,7 +79,7 @@ class PageOverlay final : public RefCounted<PageOverlay> {
     WEBCORE_EXPORT PageOverlayController* controller() const;
 
-    typedef uint64_t PageOverlayID;
+    using PageOverlayID = WebCore::PageOverlayID;
```
It seems silly to keep the `PageOverlay::PageOverlayID` type now that `WebCore::PageOverlayID` exists, but that could be a separate cleanup.

darinadler replied:
This comes up repeatedly in these types of patches. We can do larger patches that resolve these, but this seems a more expedient approach. I am willing to take it out later.

brentfulgham replied:
Agreed. I don't suggest making any further changes in this patch; I just think it would be nice to clean these up someday.

darinadler replied:
I’m also not sure that it’s all that helpful to have a `typedef` here that is just `uint64_t` with no type checking. We have been using type safe IDs instead in newer code.

brentfulgham commented on Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIEvent.h:
```diff
@@ -31,7 +31,6 @@
 #include "JSWebExtensionWrappable.h"
 #include "WebExtensionAPIObject.h"
 #include "WebExtensionEventListenerType.h"
-#include "WebPage.h"
```
:-)

brentfulgham commented on Source/WebKit/WebProcess/WebPage/DrawingArea.h:
```diff
@@ -25,48 +25,46 @@
-
-namespace WTF {
-class MachSendRight;
-}
```
This is a significant cleanup!

brentfulgham reviewed (approved):
It looks like we might still have a WinCairo build issue, but I think this change looks great.

I just had a question about replacing `const Object&` with `span` (if we are fully retaining `const` semantics).
